js/components/discoveryPanel.js

```diff
@@ -28,9 +26,39 @@ var DiscoveryPanel = React.createClass({
         })
       }),
       icons: React.PropTypes.objectOf(React.PropTypes.object)
+    }),
```

Comment:
Let's change the controller type to be object to avoid confusion with external developers.